### PR TITLE
Added wrap-doc methods #5

### PR DESCRIPTION
**Pull request by rahul799**
# Description

Please include a summary of the change and which issue is fixed.

# Checklist:
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes. (Tick if no updates were necessary.)

# How can I get additional information on failed tests during CI:
If your PR is failing you can check out 
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. If you click in the column that lists the failed tests you will get detailed error messages.

# Note:
- Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd [hunk 1]
```diff
@@ -16,14 +16,52 @@ cdef extern from "<OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h>" namespace "OpenMS":
...
+                #
+                #   The intermediate summed spectra and picked peaks can be saved to the filesystem. 
+                #   Also, the results of the accurate mass search and the signal-to-noise information 
+                #   of the resulting spectrum is saved.
+                #   -----
+                #   :param experiment: Input MSExperiment
+                #   :param n_seconds: Input number of seconds
+                #   :param load_cached_spectrum: Load the cached picked spectrum if exists
+                #   :param output: [out] Output of the accurate mass search results
+                #   :return: a boolean indicating if the picked spectrum was loaded from the cached file
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:40]: Is there a difference between when to use `:return:` and when `:returns:`. I have seen both in reST format examples?

File: src/openms/include/OpenMS/ANALYSIS/ID/IDDecoyProbability.h [hunk 2]
```diff
@@ -71,7 +71,7 @@ namespace OpenMS
     /// assignment operator
     IDDecoyProbability & operator=(const IDDecoyProbability & rhs);
 
-    /**	converts the forward and reverse identification into probabilities
+    /**	@brief converts the forward and reverse identification into probabilities
```
Comment (oliveralka) [inline src/openms/include/OpenMS/ANALYSIS/ID/IDDecoyProbability.h:74]: Could you start with a capital letter here? -> Converts

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd [hunk 3]
```diff
@@ -16,14 +16,52 @@ cdef extern from "<OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h>" namespace "OpenMS":
         #
         # wrap-inherits:
         #  DefaultParamHandler
-        FIAMSDataProcessor() nogil except +
-        FIAMSDataProcessor(FIAMSDataProcessor) nogil except +
+        FIAMSDataProcessor() nogil except + # wrap-doc:Data processing for FIA-MS data
+        FIAMSDataProcessor(FIAMSDataProcessor) nogil except + # wrap-ignore
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:20]: IMHO `# wrap-ignore `should not be needed here.

@hroest When is a # wrap-ignore needed for the copy constructor? Or is there no specific rule?

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd [hunk 4]
```diff
@@ -16,14 +16,52 @@ cdef extern from "<OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h>" namespace "OpenMS":
...
+                #   - the accurate mass search for all the picked peaks is performed
+                #
+                #   The intermediate summed spectra and picked peaks can be saved to the filesystem. 
+                #   Also, the results of the accurate mass search and the signal-to-noise information 
+                #   of the resulting spectrum is saved.
+                #   -----
+                #   :param experiment: Input MSExperiment
+                #   :param n_seconds: Input number of seconds
+                #   :param load_cached_spectrum: Load the cached picked spectrum if exists
+                #   :param output: [out] Output of the accurate mass search results
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:39]: You can remote the "[out]" here

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd (hunk 1 above)
Comment (oliveralka) [reply src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:40]: Maybe again start with a capital letter.

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd [hunk 5]
```diff
@@ -16,14 +16,52 @@ cdef extern from "<OpenMS/ANALYSIS/ID/FIAMSDataProcessor.h>" namespace "OpenMS":
...
+                #   :return: a boolean indicating if the picked spectrum was loaded from the cached file
         # void cutForTime(MSExperiment & experiment, float & n_seconds, libcpp_vector[ MSSpectrum ] & output) nogil except +
         # NAMESPACE # MSSpectrum mergeAlongTime(libcpp_vector[ OpenMS::MSSpectrum ] & input_) nogil except +
         MSSpectrum extractPeaks(MSSpectrum & input_) nogil except +
+            # wrap-doc:
+                #   Pick peaks from the summed spectrum
+                #   -----
+                #   :param input: Input vector of spectra
+                #   :return: a spectrum with picked peaks
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:48]: Capital letter? Or do you think it looks better like this? Please also see the other `:return: `statements.

File: src/pyOpenMS/pxds/FileHandler.pxd [hunk 6]
```diff
@@ -26,11 +26,46 @@ cdef extern from "<OpenMS/FORMAT/FileHandler.h>" namespace "OpenMS":
         FileHandler(FileHandler) nogil except +
 
         bool loadExperiment(String, MSExperiment &) nogil except+
+            # wrap-doc:
+            #   Loads a file into an MSExperiment
+            #   -----
+            #   :param filename: The file name of the file to load.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/FileHandler.pxd:32]: remove the dot from the end of the lines to stay consistent.

File: src/pyOpenMS/pxds/FileHandler.pxd [hunk 7]
```diff
@@ -26,11 +26,46 @@ cdef extern from "<OpenMS/FORMAT/FileHandler.h>" namespace "OpenMS":
...
+            #   :param filename: The name of the file to store the data in.
+            #   :param exp: The experiment to store.
+            #   :param log: Progress logging mode
+            #   :raises:
+            #     Exception: UnableToCreateFile is thrown if the file could not be written
         bool loadFeatures(String, FeatureMap &) nogil except +
+            # wrap-doc:
+            #   Loads a file into a FeatureMap
+            #   -----
+            #   :param filename: the file name of the file to load.
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/FileHandler.pxd:58]: Capital letter. Please also check other occurrences.

File: src/pyOpenMS/pxds/Fitter1D.pxd [hunk 8]
```diff
@@ -13,8 +13,8 @@ cdef extern from "<OpenMS/TRANSFORMATIONS/FEATUREFINDER/Fitter1D.h>" namespace "
     cdef cppclass Fitter1D(DefaultParamHandler):
         # wrap-inherits:
         #  DefaultParamHandler
-        Fitter1D() nogil except +
-        Fitter1D(Fitter1D) nogil except +
+        Fitter1D() nogil except + # wrap-doc:Abstract base class for all 1D-dimensional model fitter
+        Fitter1D(Fitter1D) nogil except + # wrap-ignore
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Fitter1D.pxd:17]: See comment to # wrap-ignore above. Let's wait for Hannes comment and then check the other instances as well.

File: src/pyOpenMS/pxds/Gradient.pxd [hunk 9]
```diff
@@ -5,38 +5,36 @@ cdef extern from "<OpenMS/METADATA/Gradient.h>" namespace "OpenMS":
...
+        void clearTimepoints() nogil except + # wrap-doc:Removes all timepoints
         # returns a reference to the list of timepoints
-        libcpp_vector[Int] getTimepoints() nogil except +
+        libcpp_vector[Int] getTimepoints() nogil except + # wrap-doc:Returns a reference to the list of timepoints
 
         #   @brief sets the percentage of eluent @p eluent at timepoint @p timepoint
-        #
         #   @exception Exception::InvalidValue is thrown if the eluent, timepoint or percentage is invalid.
-        void setPercentage(String eluent, Int timepoint, UInt percentage) nogil except +
+        void setPercentage(String eluent, Int timepoint, UInt percentage) nogil except + # wrap-doc:Brief sets the percentage of eluent @p eluent at timepoint @p timepoint
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/Gradient.pxd:30]: remove @p and correct the senctence.

File: src/pyOpenMS/pxds/GridBasedCluster.pxd [hunk 10]
```diff
@@ -17,16 +17,16 @@ cdef extern from "<OpenMS/COMPARISON/CLUSTERING/GridBasedCluster.h>" namespace "
...
         GridBasedCluster(DPosition2 centre, 
            DBoundingBox2 bounding_box,
            libcpp_vector[ int ] point_indices) nogil except +
 
-        DPosition2 getCentre() nogil except +
-        DBoundingBox2 getBoundingBox() nogil except +
-        libcpp_vector[ int ] getPoints() nogil except +
-        int getPropertyA() nogil except +
-        libcpp_vector[ int ] getPropertiesB() nogil except +
+        DPosition2 getCentre() nogil except + Returns cluster centre
```
Comment (oliveralka) [inline src/pyOpenMS/pxds/GridBasedCluster.pxd:25]: Missing `# wrap-doc` (whole file)

**Review by oliveralka — CHANGES_REQUESTED** [review]
Looks good so far! Needs some minor changes.

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd (hunk 1 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:40]: No difference. I have left 'a' small intentionally.

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd (hunk 3 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:20]: @oliveralka should I remove it?

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd (hunk 5 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:48]: yeah, it looks better. I left **'a'** small intentionally 😃

File: src/pyOpenMS/pxds/Fitter1D.pxd (hunk 8 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/Fitter1D.pxd:17]: ok

File: src/pyOpenMS/pxds/GridBasedCluster.pxd (hunk 10 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/GridBasedCluster.pxd]: Fixed 😄 .

File: src/pyOpenMS/pxds/FileHandler.pxd (hunk 7 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/FileHandler.pxd]: Fixed.

File: src/pyOpenMS/pxds/FileHandler.pxd (hunk 6 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/FileHandler.pxd]: Fixed

File: src/pyOpenMS/pxds/Gradient.pxd (hunk 9 above)
Comment (rahul799) [reply src/pyOpenMS/pxds/Gradient.pxd]: Done

File: src/pyOpenMS/pxds/FIAMSDataProcessor.pxd (hunk 3 above)
Comment (oliveralka) [reply src/pyOpenMS/pxds/FIAMSDataProcessor.pxd:20]: As far as I have seen the  `# wrap-ignore` for the copy constructor is also automatically added in the `PythonExtensionChecker.py`. So IMHO, let's leave the # wrap-ignore for now. Any objections @timosachsenberg @hroest

**Review by oliveralka — APPROVED** [review]
Looks good!

**Comment by oliveralka** [comment]
rebuild jenkins

**Pull request merged**